server/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -701,6 +702,14 @@ private void createShard(ShardRouting shardRouting, ClusterState state) {
                     logger.trace("ignoring initializing shard {} - no source node can be found.", shardId);
                     return;
                 }
+            } else if (shardRouting.recoverySource().getType() == Type.RESHARD_SPLIT_TARGET) {
```

Comment:
not a request for change, this isn't your API, but it would be nice if the ShardRouting API didn't make us get type and then cast in two steps. I could imagine something like a `shardRouting.recoverySource().asReshardSplit()` maybe that did the check and returned the cast object if it passed or null.

I'm just griping though, it's probably not worth doing now.

Reply:
I could `instanceof` the `shardRouting.recoverySource()` instead. What do you think?